frontend/src/Components/ProtectedRoutes/ProtectedRoutes.jsx

```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import { Route, Navigate } from "react-router-dom";
+
+const ProtectedRoute = ({ element: Component, ...rest }) => {
+  const isAuthenticated =
+    localStorage.getItem("user") && localStorage.getItem("token");
```

Comment:
maybe you plan on modifying this in a later PR, but this isn't a good authentication check.  Any user can put a dummy value in their local storage such that `isAuthenticated` becomes true.  Ideally we actually somehow validate that the token's value is a valid, non-expired token.

Reply:
I would love some more clarity on this if you don't mind. And how would i would go about doing this